spring-test/src/main/java/org/springframework/test/web/client/match/ContentRequestMatchers.java

```diff
@@ -363,7 +365,9 @@ private static class MultipartHelper {
 		public static MultiValueMap<String, ?> parse(MockClientHttpRequest request) {
 			try {
 				FileUpload fileUpload = new FileUpload();
-				fileUpload.setFileItemFactory(new DiskFileItemFactory());
+				DiskFileItemFactory factory = new DiskFileItemFactory();
+				factory.setDefaultCharset(DEFAULT_ENCODING);
+				fileUpload.setFileItemFactory(factory);
```

Comment:
Switching the general default could break existing tests, and that should wait for a major or minor release. In the meantime, we could narrow the scope and default to UTF-8 specifically for JSON. That would be safer and addresses the original issue.